src/main/java/com/example/demo/categories/CategoryRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.example.demo.categories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Optional;
+
+public interface CategoryRepository extends JpaRepository<Category, Integer> {
+
+    @Query("SELECT c FROM Category c WHERE c.name = ?1")
+    Optional<Category> findCategoryByName(String name);
```

Comment:
For this we dont need an @query annotation. Hibernate already provides the findById() method by default